packages/textlint-tester/README.md

```diff
@@ -11,20 +11,91 @@
 1. Write tests by using `textlint-tester`
 2. Run tests by [Mocha](http://mochajs.org/ "Mocha")
 
+```sh
+$(npm bin)/mocha test/
+```
+
 ### TextLintTester
 
+`TextLintTester#run` has two signatures.
+
+- If you want to test single rule, use first method (`TextLintTester#run(ruleName, rule, {valid=[], invalid=[]})`)
+- If you want to test multiple rules and/or plugins, use second method (`TextLintTester#run(testName, testConfig, {valid=[], invalid=[]})`)
+
 #### TextLintTester#run(ruleName, rule, {valid=[], invalid=[]})
 
+- `{string} ruleName` ruleName is a name of the rule.
+- `{TextlintRuleCreator} rule` rule is textlint rule.
+
+#### TextLintTester#run(testName, testConfig, {valid=[], invalid=[]})
+
+- `{string} testName` testName is a name of the test.
+- `{TestConfig} testConfig` testConfig is the configuration object for the test.
+
+##### TestConfig object
+
+`TestConfig` is defined as follows:
+
+```typescript
+export declare type TestConfig = {
+    plugins?: {
+        pluginId: string; // name of plugin
+        plugin: TextlintPluginCreator; // textlint plugin
+        options?: any; // options for plugin
+    }[];
+    rules: {
+        ruleId: string; // name of rule
+        rule: TextlintRuleCreator; // textlint rule
+        options?: any; // optionsfor rule
+    }[];
+};
+```
+
+`testConfig` object example:
+
+```js
+{
+    plugins: [
+        {
+            pluginId: "html",
+            plugin: htmlPlugin // = require("textlint-plugin-html")
+        }
+    ],
+    rules: [
+        {
+            ruleId: "no-todo",
+            rule: noTodoRule // = require("textlint-rule-no-todo")
+        },
+        {
+            ruleId: "max-number-of-lines",
+            rule: maxNumberOfLineRule, // = require("textlint-rule-max-number-of-lines")
+            options: {
+                max: 2
+            }
+        }
+    ]
+}
+```
+
 ##### valid object
 
-- `{string} ruleName` ruleName is a name of the rule.
-- `{Function} rule` rule is the exported function of the rule.
 - `{string[]|object[]} valid` valid is an array of text which should be passed.
     - You can use `object` if you want to specify some options. `object` can have the following properties:
         - `{string} text`: a text to be linted
         - `{string} ext`: an extension key. Default: `.md` (Markdown)
         - `{string} inputPath`: a test text filePath that prefer to `text` property
-        - `{object} options`: options to be passed to the rule
+        - `{object} options`: options to be passed to the rule. Will be ignored if `testConfig` is specified 
```

Comment:
> Will be ignored if `testConfig` is specified 

1. if specified `testConfig` property in `valid` object, ignore `options` property".
2. if specified `testConfig` parameter in `run` method, ignore `options` property".

Which is correct?
I think that both is correct, but we need to write clearly.

`1` means that you can override `run`'s `testConfig` parameter by `valid`'s `testConfig` property.
`2` means that you can not override `run`'s `testConfig` parameter in each test case.

`2` have more restriction than  `1`.
Which is better? It is case by case, I think.

Reply:
We need to add an assertion for invalid pattern after this.

Reply:
@azu, thanks a ton for your continuous review.

What I meant is 2. It should be good enough for now. Overriding gives test case writer more flexibility but writing configuration for multiple rules could be complex. I think we can implement option 1 if we found such use case. I will work on 2 first. As you said need more clear explanation so will update README as well.

Reply:
Yes. I agree with you.
`2` is enough as a first step.